vnext/Shared/Threading/BatchingQueueThread.cpp

```diff
@@ -57,9 +54,48 @@ void BatchingQueueThread::PostBatch() noexcept {
   }
 }
 
+void BatchingQueueCallInvoker::onBatchComplete() noexcept {
+  PostBatch();
+}
+
+void BatchingQueueCallInvoker::quitSynchronous() noexcept {
+  PostBatch();
+  m_queueThread->quitSynchronous();
+}
+
+void BatchingQueueCallInvoker::invokeSync(std::function<void()> &&func) noexcept {
+  assert(false && "Not supported");
+  std::terminate();
+}
+
+BatchingQueueThread::BatchingQueueThread(
+    std::shared_ptr<facebook::react::MessageQueueThread> const &queueThread) noexcept {
+  m_batchingQueueCallInvoker = std::make_shared<BatchingQueueCallInvoker>(queueThread);
+  m_callInvoker = m_batchingQueueCallInvoker;
+}
+
+// Called from the JS thread, once the instance has had a chance to setup the NativeToJsBridge and calls to
+// getDecoratedNativeCallInvoker will work.
+void BatchingQueueThread::decoratedNativeCallInvokerReady(
+    std::weak_ptr<facebook::react::Instance> wkInstance) noexcept {
+  std::scoped_lock lck(m_mutex);
+  m_callInvoker->invokeAsync([wkInstance, this] {
+    if (auto instance = wkInstance.lock()) {
+      m_callInvoker = instance->getDecoratedNativeCallInvoker(m_callInvoker);
```

Comment:
>m_callInvoker [](start = 6, length = 13)

We neeed to add the `std::scoped_lock lck(m_mutex);` inside of this lambda to protect the write access to the m_callInvoker. So that readers of the m_callInvoker can access it only after it completely changed.